dlordinal/layers/clm.py

```diff
@@ -45,13 +44,11 @@ def _convert_thresholds(self, b, a, min_distance):
         thresholds_param = torch.cat((b, a), dim=0)
         th = torch.sum(
             torch.tril(
-                torch.ones((self.num_classes - 1, self.num_classes - 1)).to(
-                    self.device
-                ),
+                torch.ones((self.num_classes - 1, self.num_classes - 1)),
```

Comment:
When you create this tensor, you have to create it in the same device that the others tensors use. You can use the `projected` variable or one of the thresholds variables.